+++ setup.py	2009-06-12 01:20:24.000000000 +0200
--- setup.py.orig	2009-06-12 01:15:46.000000000 +0200
@@ -68,7 +68,7 @@
                 raise SystemError, "Nvidia's CUDA-compiler 'nvcc' can't be found. Make sure it's available to $PATH. " \
                                     "It is part of the CUDA Toolkit (not the SDK)."
             print "Compiling CUDA module using nvcc %s..." % nvcc_version
+            subprocess.check_call(NVCC + ' --host-compilation C --opencc-options -WOPT:expr_reass=off -Xptxas "-v" -Xcompiler "-fPIC" --cubin ./_cpyrit_cudakernel.cu', shell=True)
-            subprocess.check_call(NVCC + ' --host-compilation C -Xptxas "-v" -Xcompiler "-fPIC" --cubin ./_cpyrit_cudakernel.cu', shell=True)
             f = open("_cpyrit_cudakernel.cubin", "rb")
             cubin_inc = ",".join(("0x%02X%s" % (ord(c), "\n" if i % 32 == 0 else "") for i, c in enumerate(f.read())))
             f.close()
