--- NVIDIA-Linux-x86-173.14.18-pkg0/usr/src/nv/nv.c.orig	2009-06-14 17:08:34.000000000 +0200
+++ NVIDIA-Linux-x86-173.14.18-pkg0/usr/src/nv/nv.c	2009-06-14 17:15:41.000000000 +0200
@@ -603,14 +603,6 @@
     if (!proc_nvidia_warnings)
         goto failed;
 
-    /*
-     * Set the module owner to ensure that the reference
-     * count reflects accesses to the proc files.
-     */
-    proc_nvidia->owner = THIS_MODULE;
-    proc_nvidia_cards->owner = THIS_MODULE;
-    proc_nvidia_warnings->owner = THIS_MODULE;
-
     for (j = 0; j < num_nv_devices; j++)
     {
         nvl = &nv_linux_devices[j];
@@ -629,7 +621,6 @@
 
         entry->data = nv;
         entry->read_proc = nv_kern_read_cardinfo;
-        entry->owner = THIS_MODULE;
 
         if (nvos_find_agp_capability(dev)) {
             /*
@@ -642,7 +633,6 @@
                 goto failed;
             }
 
-            entry->owner = THIS_MODULE;
             proc_nvidia_agp = entry;
 
             entry = create_proc_entry("status", flags, proc_nvidia_agp);
@@ -653,7 +643,6 @@
 
             entry->data = nv;
             entry->read_proc = nv_kern_read_status;
-            entry->owner = THIS_MODULE;
 
             entry = create_proc_entry("host-bridge", flags, proc_nvidia_agp);
             if (!entry) {
@@ -663,7 +652,6 @@
 
             entry->data = NULL;
             entry->read_proc = nv_kern_read_agpinfo;
-            entry->owner = THIS_MODULE;
 
             entry = create_proc_entry("card", flags, proc_nvidia_agp);
             if (!entry) {
@@ -673,7 +661,6 @@
 
             entry->data = nv;
             entry->read_proc = nv_kern_read_agpinfo;
-            entry->owner = THIS_MODULE;
         }
 
         NV_PCI_DEV_PUT(dev);
@@ -684,14 +671,12 @@
         goto failed;
 
     entry->read_proc = nv_kern_read_version;
-    entry->owner = THIS_MODULE;
 
     entry = create_proc_entry("registry", flags, proc_nvidia);
     if (!entry)
         goto failed;
 
     entry->read_proc = nv_kern_read_registry;
-    entry->owner = THIS_MODULE;
 
     return;
 
@@ -718,7 +703,6 @@
 
     entry->data = (void *)message;
     entry->read_proc = nv_kern_read_warning;
-    entry->owner = THIS_MODULE;
 #endif
 }
 
